src/trg-client.c

```diff
@@ -82,14 +82,17 @@ struct _TrgClientPrivate {
     GHashTable *torrentTable;
     GThreadPool *pool;
     TrgPrefs *prefs;
-    GPrivate tlsKey;
     gint configSerial;
     guint http_class;
     GMutex configMutex;
     gboolean seedRatioLimited;
     gdouble seedRatioLimit;
 };
 
+static void trg_tls_free(gpointer data);
+
+static GPrivate tls_key = G_PRIVATE_INIT(trg_tls_free);
```

Comment:
Reason for making this a global?

Reply:
Perhaps I misunderstood the documentation, but I read it as GPrivate should always be static and any other use is deprecated. Also I think it was a requirement for using G_PRIVATE_INIT, but I might have got that wrong..

Reply:
I say leave it alone for now.

If you want to make the code modern get rid of `GPrivate` entirely, thread-local-storage has been a standard part of C for decades now.

Reply:
I wanted to add a cleanup method, trg_tls_free, to release any resources allocated in the TLS, and this was the only method I found

Reply:
It just changes a per-instance property into a global shared one. I realize that doesn't matter in practice but its just wrong.

Reply:
Isn't an instance equal to a pool thread in this case? I understood it as that a GPrivate is a property of the current thread, and it will have different values depending on which thread you try to access it from. (Assuming I got that correct) I thought it was different enough from the other instance variables to warrant special treatment.

But I don't want to waste your time on meaningless discussions, so I'll revert this part if you think it was better before. The problem I tried to solve was that pool threads don't free their resources when destroyed, and maybe that can be done in some other way or isn't necessary at all.

Reply:
Removed related changes from this commit

Reply:
As I said, I think its possible to move away from `GPrivate` entirely.

This is C11:

```
#include <threads.h>
thread_local int foo = 0;
```

Which does mean dropping support for MSVC on Windows but... I don't know that anybody cares.

Reply:
Sorry, I missed your last comments entirely. I think the reason for this TLS is for worker threads to reuse existing curl connections, but maybe it's possible to do it without too.

That thread_local variant sure looks clean, and if it comes with a way to run a custom free-function when the thread exits it could probably be used instead.

However, I submitted my proposed change (i.e. static GPrivate and added free function) if you want a fix for the missing cleanup on worker thread exit until a better solution is implemented.